tests/test_binary.nim

```diff
@@ -271,13 +271,12 @@ proc main =
   const helpStart = &"Usage:\n  {binaryName} [global-options] <command> [command-options]"
 
   const cmd = "nimble --verbose build -d:release"
-  stdout.write(&"Running `{cmd}`... ")
-  stdout.flushFile()
+  stderr.write(&"Running `{cmd}`... ")
```

Comment:
Why is this writing to `stderr`?

Reply:
The purpose of this line is to tell the user that something is happening (otherwise it appears like a hang), and I think it's standard for such messaging to go to `stderr`.

For example, if we make a request with `curl` and redirect stdout
```
curl https://releases.ubuntu.com/20.04.2.0/ubuntu-20.04.2.0-desktop-amd64.iso > /dev/null
```
we see that progress is printed to `stderr`.

Using `stderr` in the Nim code also avoids a `flushFile`, since `stderr` is unbufferred by default.

But it's somewhat arbitrary - everything else from running the tests is written to `stdout`. That is, with the latest commit on `main`, if you compile `tests/all_tests.nim` and then run
```shell
$ ./all_tests > /dev/null
``` 
nothing is printed to the screen even if the tests fail. That's because Nim's `unittest` module takes the position that all test results are "output" and so should appear on `stdout`.

By changing to `stderr` here, the user who writes
```shell
$ ./all_tests > /tmp/configlet_tests.txt
```
will no longer see the diagnostic message in that file, only the test results. I think that's a little better.

Can we just do `stderr` here for now?

Reply:
Yeah that makes sense. I was mostly just curious.